framework/include/interfacekernels/InterfaceReaction.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InterfaceKernel.h"
+
+// Forward Declarations
+class InterfaceReaction;
+
+template <>
+InputParameters validParams<InterfaceReaction>();
+
+class InterfaceReaction : public InterfaceKernel
```

Comment:
Please add doxygen here for what this class does. See the `Diffusion` kernel for an example of this.